vllm/entrypoints/openai/rpc/server.py

```diff
@@ -41,78 +42,66 @@ def cleanup(self):
+            elif request == RPCUtilityRequest.GET_SCHEDULER_CONFIG:
+                config = await self.engine.get_scheduler_config()
+            elif request == RPCUtilityRequest.GET_PARALLEL_CONFIG:
+                config = await self.engine.get_parallel_config()
+            else:
+                raise ValueError("Unknown Config Request: %s", request)
 
-    async def get_parallel_config(self, identity):
-        """Send the ParallelConfig"""
-        parallel_config = await self.engine.get_parallel_config()
+            await self.socket.send_multipart(
+                [rpc_id, client_id,
+                 cloudpickle.dumps(config)])
 
-        await self.socket.send_multipart(
-            [identity, cloudpickle.dumps(parallel_config)])
+        except Exception as e:
+            await self.socket.send_multipart(
+                [rpc_id, client_id, cloudpickle.dumps(e)])
 
-    async def is_tracing_enabled(self, identity):
+    async def is_tracing_enabled(self, rpc_id, client_id):
         """Send the is_tracing_enabled flag"""
         tracing_flag = await self.engine.is_tracing_enabled()
 
         await self.socket.send_multipart(
-            [identity, cloudpickle.dumps(tracing_flag)])
+            [rpc_id, client_id,
+             cloudpickle.dumps(tracing_flag)])
 
-    async def do_log_stats(self, identity):
+    async def do_log_stats(self, rpc_id, client_id):
         """Log stats and confirm success."""
         await self.engine.do_log_stats()
 
-        await self.socket.send_multipart([
-            identity,
-            cloudpickle.dumps(VLLM_RPC_SUCCESS_STR),
-        ])
+        await self.socket.send_multipart(
+            [rpc_id, client_id,
+             cloudpickle.dumps(VLLM_RPC_SUCCESS_STR)])
 
-    async def is_server_ready(self, identity):
+    async def is_server_ready(self, rpc_id, client_id):
         """Notify the client that we are ready."""
-        await self.socket.send_multipart([
-            identity,
-            cloudpickle.dumps(VLLM_RPC_SUCCESS_STR),
-        ])
+        await self.socket.send_multipart(
+            [rpc_id, client_id,
+             cloudpickle.dumps(VLLM_RPC_SUCCESS_STR)])
 
-    async def abort(self, identity, request: RPCAbortRequest):
+    async def abort(self, rpc_id, client_id, request: RPCAbortRequest):
         """Abort request and notify the client of success."""
         try:
             # Abort the request in the llm engine.
             await self.engine.abort(request.request_id)
-        except Exception:
-            logger.warning("Failed to abort request %s", request.request_id)
-        finally:
-            # Send confirmation to the client.
-            await self.socket.send_multipart([
-                identity,
-                cloudpickle.dumps(VLLM_RPC_SUCCESS_STR),
-            ])
-
-    async def generate(self, identity, generate_request: RPCGenerateRequest):
+            await self.socket.send_multipart(
+                [rpc_id, client_id,
```

Comment:
note: if there is an exception calling `abort`, we should propagate this to the client

Reply:
👍 , makes sense